sn_build_info/src/lib.rs

```diff
@@ -33,3 +34,58 @@ pub const fn git_branch() -> &'static str {
 pub const fn git_sha() -> &'static str {
     env!("VERGEN_GIT_SHA")
 }
+
+/// Nightly version format: YYYY.MM.DD
+pub fn nightly_version() -> String {
+    let now = Utc::now();
+    now.format("%Y.%m.%d").to_string()
+}
+
+/// Git information for nightly builds: `<date> / <branch> / <sha>`
+pub fn nightly_git_info() -> String {
+    format!("{} / {} / {}", nightly_version(), git_branch(), git_sha(),)
+}
+
+pub fn package_version() -> String {
+    format!(
+        "{}.{}.{}.{}",
+        env!("RELEASE_YEAR"),
+        env!("RELEASE_MONTH"),
+        env!("RELEASE_CYCLE"),
+        env!("RELEASE_CYCLE_COUNTER")
+    )
+}
+
+pub fn full_version_info(crate_version: &str, protocol_version: &str) -> String {
+    format!(
+        "v{}\nNetwork version: {}\nPackage version: {}\nGit info: {}",
+        crate_version,
+        protocol_version,
+        package_version(),
+        git_info()
+    )
+}
+
+pub fn version_string(crate_version: &str, protocol_version: &str) -> String {
+    if cfg!(feature = "nightly") {
+        format!(
+            "-- Nightly Release {}\nGit info: {} / {}",
+            nightly_version(),
+            git_branch(),
+            git_sha(),
+        )
+    } else {
+        full_version_info(crate_version, protocol_version)
+    }
+}
+
+pub fn log_version_info(crate_version: &str, protocol_version: &str) {
+    if cfg!(feature = "nightly") {
+        debug!("nightly build info: {}", nightly_git_info());
```

Comment:
shoudl nightly also display the `protocol_version`?

Reply:
Can we also add the following here?
```
  debug!("version: {crate_version}");
  debug!("network version: {protocol_version}");
```

Reply:
The version (the date) is in the nightly build info. However, I added the network version line.